pkg/assembler/backends/neo4j/certifyVEXStatement.go

```diff
@@ -17,11 +17,437 @@ package neo4jBackend
 )
 
 func (c *neo4jClient) CertifyVEXStatement(ctx context.Context, certifyVEXStatementSpec *model.CertifyVEXStatementSpec) ([]*model.CertifyVEXStatement, error) {
-	panic(fmt.Errorf("not implemented: CertifyVEXStatement - CertifyVEXStatement"))
+	querySubjectAll := false
+	if certifyVEXStatementSpec.Package != nil && certifyVEXStatementSpec.Artifact != nil {
+		return nil, gqlerror.Errorf("cannot specify package and artifact together for CertifyVEXStatement")
+	} else if certifyVEXStatementSpec.Package == nil && certifyVEXStatementSpec.Artifact == nil {
+		querySubjectAll = true
+	}
+
+	queryVulnAll := false
+	if certifyVEXStatementSpec.Cve != nil && certifyVEXStatementSpec.Ghsa != nil {
+		return nil, gqlerror.Errorf("cannot specify cve and ghsa together for CertifyVEXStatement")
+	} else if certifyVEXStatementSpec.Cve == nil && certifyVEXStatementSpec.Ghsa == nil {
+		queryVulnAll = true
+	}
+
+	queryAll := false
+	if querySubjectAll && queryVulnAll {
+		queryAll = true
+	}
+
+	session := c.driver.NewSession(neo4j.SessionConfig{AccessMode: neo4j.AccessModeRead})
+	defer session.Close()
+
+	aggregateCertifyVEXStatement := []*model.CertifyVEXStatement{}
+
+	if queryAll || querySubjectAll && certifyVEXStatementSpec.Cve != nil || queryVulnAll && certifyVEXStatementSpec.Package != nil ||
+		certifyVEXStatementSpec.Package != nil && certifyVEXStatementSpec.Cve != nil {
+
+		var sb strings.Builder
+		var firstMatch bool = true
+		queryValues := map[string]any{}
+
+		// query CVE
+		returnValue := " RETURN type.type, namespace.namespace, name.name, version.version, version.subpath, " +
+			"version.qualifier_list, certifyVEXStatement, cveYear.year, cveID.id"
+
+		// query with pkgVersion
+		query := "MATCH (root:Pkg)-[:PkgHasType]->(type:PkgType)-[:PkgHasNamespace]->(namespace:PkgNamespace)" +
+			"-[:PkgHasName]->(name:PkgName)-[:PkgHasVersion]->(version:PkgVersion)" +
+			"-[:subject]-(certifyVEXStatement:CertifyVEXStatement)-[:about]-(cveID:CveID)<-[:CveHasID]" +
+			"-(cveYear:CveYear)<-[:CveIsYear]-(rootCve:Cve)"
+		sb.WriteString(query)
+
+		setPkgMatchValues(&sb, certifyVEXStatementSpec.Package, false, &firstMatch, queryValues)
+		setCveMatchValues(&sb, certifyVEXStatementSpec.Cve, &firstMatch, queryValues)
+		setCertifyVEXStatementValues(&sb, certifyVEXStatementSpec, &firstMatch, queryValues)
+		sb.WriteString(returnValue)
+
+		if certifyVEXStatementSpec.Package == nil || certifyVEXStatementSpec.Package != nil && certifyVEXStatementSpec.Package.Version == nil &&
+			certifyVEXStatementSpec.Package.Subpath == nil && len(certifyVEXStatementSpec.Package.Qualifiers) == 0 &&
+			!*certifyVEXStatementSpec.Package.MatchOnlyEmptyQualifiers {
+
+			sb.WriteString("\nUNION")
+			// query without pkgVersion
+			query = "\nMATCH (root:Pkg)-[:PkgHasType]->(type:PkgType)-[:PkgHasNamespace]->(namespace:PkgNamespace)" +
+				"-[:PkgHasName]->(name:PkgName)" +
+				"-[:subject]-(certifyVEXStatement:CertifyVEXStatement)-[:about]-(cveID:CveID)<-[:CveHasID]" +
```

Comment:
same comment but also just putting a comment since there's a chain of PRs, so let you close comments in order of merging.